src/Microsoft.InsightsGenerator/DataTransformation.cs

```diff
@@ -3,13 +3,132 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
+using System;
+using System.Collections.Generic;
+
 namespace Microsoft.InsightsGenerator
 {
     public class DataTransformer
     {
+        private class ColumnInfo
+        {
+            public int ColumnIndex { get; set; }
+
+            public int DistinctValues { get; set; }
+
+            public DataArray.DataType DataType { get; set; }
+        }
+
         public DataArray Transform(DataArray array)
         {
+            if (array == null || array.Cells == null && array.Cells.Length > 0)
+            {
+                return array;
+            }
+
+            array.ColumnNames = GetColumnLabels(array);
             return array;
-        } 
+        }
+
+        private string[] GetColumnLabels(DataArray array)
+        {
+            int columnCount = array.Cells[0].Length;
+            Dictionary<DataArray.DataType, List<ColumnInfo>> columnInfo = new Dictionary<DataArray.DataType, List<ColumnInfo>>();
+            for (int column = 0; column < columnCount; ++column)
+            {
+                int distinctValues;
+                DataArray.DataType dataType = GetColumnType(array, column, out distinctValues);
+                if (!columnInfo.ContainsKey(dataType))
+                {
+                    columnInfo.Add(dataType, new List<ColumnInfo>());
+                }
+
+                columnInfo[dataType].Add(new ColumnInfo()
+                {
+                    ColumnIndex = column,
+                    DistinctValues = distinctValues,
+                    DataType = dataType
+                });
+            }
+
+            bool containsDateTime = columnInfo.ContainsKey(DataArray.DataType.DateTime);
+            string[] labels = new string[columnCount];
+            if (containsDateTime)
+            {
+                List<ColumnInfo> dateColumns = columnInfo[DataArray.DataType.DateTime];
+                for (int i = 0; i < dateColumns.Count; ++i)
+                {
+                    labels[dateColumns[i].ColumnIndex] = "input_t_" + i;
+                }
+            }
+
+            if (columnInfo.ContainsKey(DataArray.DataType.String))
+            {
+                int startingIndex = 0;
```

Comment:
this code block should assign the input_g label to the column with the lowest distinctValue count

Reply:
Here are some of the notes on what the expect output column names would be in various cases:
```
Input
 - Time
 - Groups
Slicer
 - 1...N
Output
 - 1..N

Time
String
Number

-> Input_t, Slicer, Output


Time
String ->input_g
String -> slicer
Number

-> Input_t, Input_g, Slicer, Output

Time -> input_t_0
String -> slicer_0
String -> slicer_1
String -> input_g
String -> slicer_2
Number -> output_0
Number -> output_1
Time -> input_t_1
```